### PR TITLE
#60717378 Edit travel team member center 

### PR DESCRIPTION
**Pull request by Spectrumsun**
#### What does this PR do?
- The travel admin can change the center of a travel team member

#### Description of Task to be completed?
- A travel admin can change the center of a travel team member

#### How should this be manually tested?
- Clone the repo - `git clone https://github.com/andela/travel_tool_back.git`
- Enter into the project directory - `cd travel_tool_back`
- Rollback, run database migrations and seed database - `yarn db:rollmigrate`
- Install dependencies and start the server - `make start`
-  [Start](https://github.com/andela/travel_tool_front/pull/107)  the frontend.
- Navigate to [travela](http://travela-local.andela.com:3000/#) in your browser and login
- Grab your token from the application tab in Developers tab
- Open `Postman`
- Add your token in headers

- Send a `PATCH` request to `http://localhost:5000/api/v1/center/user/{userId}` 
- With center as the name of the field and the new center you want to change it to

#### Any background context you want to provide?
- You have to be a travel admin or a super admin before you can change the center of a travel team member
- The user must have a role of travel team member 

#### What are the relevant pivotal tracker stories?
[160717378](https://www.pivotaltracker.com/story/show/160717378l)

#### Screenshots (if appropriate)
image

#### Questions:
- none

### CODE REVIEW

File: src/modules/guestHouse/__tests__/getGethouses.test.js [hunk 1]
```diff
@@ -31,16 +39,24 @@ describe('Get Guest Houses', () => {
       });
   });
 
+  afterAll(async () => {
```
Comment (albert85) [inline src/modules/guestHouse/__tests__/getGethouses.test.js]: @Spectrumsun add `done` to this

File: src/modules/guestHouse/__tests__/getGuestHouseDetails.test.js [hunk 2]
```diff
@@ -41,10 +43,20 @@ const createTestGuestHouse = data => (
 );
 
 describe('Guest house details test', () => {
-  beforeAll(() => {
+  beforeAll(async () => {
```
Comment (albert85) [inline src/modules/guestHouse/__tests__/getGuestHouseDetails.test.js]: @Spectrumsun add `done` to this

File: src/modules/guestHouse/__tests__/getGuestHouseDetails.test.js [hunk 3]
```diff
@@ -41,10 +43,20 @@ const createTestGuestHouse = data => (
...
+    await models.Role.sync({ force: true });
+    await models.Role.bulkCreate(role);
+    await models.User.sync({ force: true });
+    await models.UserRole.destroy({ force: true, truncate: { cascade: true } });
     process.env.DEFAULT_ADMIN = 'john.snow@andela.com';
   });
 
+  afterAll(async () => {
```
Comment (albert85) [inline src/modules/guestHouse/__tests__/getGuestHouseDetails.test.js]: @Spectrumsun add `done` to this

File: src/modules/guestHouse/__tests__/updateGuestHouse.test.js [hunk 4]
```diff
@@ -32,17 +36,23 @@ describe('Update the room fault status', () => {
         done();
       });
   });
+  afterAll(async () => {
```
Comment (albert85) [inline src/modules/guestHouse/__tests__/updateGuestHouse.test.js]: @Spectrumsun add `done` to this

File: src/modules/guestHouse/__tests__/updateRoomStatus.test.js [hunk 5]
```diff
@@ -23,32 +24,44 @@ const invalidToken = 'YYTRYIM0nrbuy7tonfenu';
...
-        userId: '-TRUniolpknbnk',
+        userId: '-TRUniplpknbbh',
       })
       .end((err) => {
         if (err) return done(err);
         done();
       });
   });
 
+  afterAll(async () => {
```
Comment (albert85) [inline src/modules/guestHouse/__tests__/updateRoomStatus.test.js]: @Spectrumsun add `done` to this

File: src/modules/requests/__tests__/includeBed.test.js [hunk 6]
```diff
@@ -40,6 +44,12 @@ describe('Get an authenticated User Request detail', () => {
         done();
       });
   });
+
+  afterAll(async () => {
```
Comment (albert85) [inline src/modules/requests/__tests__/includeBed.test.js]: @Spectrumsun add `done` to this

File: src/modules/userRole/__tests__/travelTeamRole.test.js [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+        userId: 'wer45660+++'
+      })
+      .expect(201)
+      .end((err) => {
+        if (err) return done(err);
+        done();
+      });
+  });
+
+  afterAll(async () => {
```
Comment (albert85) [inline src/modules/userRole/__tests__/travelTeamRole.test.js]: @Spectrumsun add `done` to this

File: src/modules/centers/__tests__/CentersController.test.js [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+let createdCenters;
+const token = Utils.generateTestToken(payload);
+const invalidToken = 'eyJhbGciOiJSUzI1NiIsXVCJ9.eyJVc2VySW5mbyI6eyJpZ';
+describe('Centers controller', () => {
+  beforeAll(async () => {
+    await models.Center.destroy({ truncate: true, cascade: true });
+    createdCenters = await models.Center.bulkCreate(centers);
+  });
+  afterAll(async () => {
+    await models.Center.destroy({ truncate: true, cascade: true });
```
[the remark this answers is not in the document]
Comment (Spectrumsun) [reply src/modules/centers/__tests__/CentersController.test.js]: it does not need done. it's an async not promise, that why its they is an await in the front. @albert85

File: src/modules/centers/__tests__/CentersController.test.js [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+    name: 'Some Tester',
+    email: 'some.tester@andela.com',
+    picture: 'fakepicture.png',
+  }
+};
+let createdCenters;
+const token = Utils.generateTestToken(payload);
+const invalidToken = 'eyJhbGciOiJSUzI1NiIsXVCJ9.eyJVc2VySW5mbyI6eyJpZ';
+describe('Centers controller', () => {
+  beforeAll(async () => {
```
[the remark this answers is not in the document]
Comment (Spectrumsun) [reply src/modules/centers/__tests__/CentersController.test.js]: it does not need done. it's an async not promise, that why its they is an await in the front. @albert85

File: src/modules/comments/__tests__/comments.test.js [hunk 10]
```diff
@@ -36,18 +36,26 @@ const token = Utils.generateTestToken(payload);
...
+        done();
+      });
   });
-  afterAll((done) => {
-    models.Role.destroy({ force: true, truncate: { cascade: true } });
-    models.User.destroy({ force: true, truncate: { cascade: true } });
-    models.Request.destroy({ force: true, truncate: { cascade: true } });
-    models.Comment.destroy({ force: true, truncate: { cascade: true } });
-    done();
+  afterAll(async () => {
```
[the remark this answers is not in the document]
Comment (Spectrumsun) [reply src/modules/comments/__tests__/comments.test.js]: same here

[the remark this answers is not in the document]
Comment (Spectrumsun) [reply src/modules/comments/__tests__/comments.test.js]: it does not need done. it's an async not promise, that why its they is an await in the front @albert85

File: src/modules/guestHouse/__tests__/editGuestHouse.test.js [hunk 11]
```diff
@@ -102,18 +107,18 @@ describe('Update Guest Houses', () => {
     });
     describe('Authenticated travel admin edits guest houses', () => {
       beforeAll(async () => {
```
[the remark this answers is not in the document]
Comment (Spectrumsun) [reply src/modules/guestHouse/__tests__/editGuestHouse.test.js]: it does not need done. it's an async not promise, that why its they is an await in the front @albert85

File: src/modules/guestHouse/__tests__/getGuestHouseDetails.test.js (hunk 3 above)
Comment (Spectrumsun) [reply src/modules/guestHouse/__tests__/getGuestHouseDetails.test.js]: it does not need done. it's an async not promise, that why its they is an await in the front @albert85

File: src/modules/guestHouse/__tests__/updateRoomStatus.test.js (hunk 5 above)
Comment (Spectrumsun) [reply src/modules/guestHouse/__tests__/updateRoomStatus.test.js]: it does not need done. it's an async not promise, that why its they is an await in the front @albert85

File: src/modules/requests/__tests__/includeBed.test.js (hunk 6 above)
Comment (Spectrumsun) [reply src/modules/requests/__tests__/includeBed.test.js]: it does not need done. it's an async not promise, that why its they is an await in the front @albert85

File: src/modules/userRole/__tests__/travelTeamRole.test.js (hunk 7 above)
Comment (Spectrumsun) [reply src/modules/userRole/__tests__/travelTeamRole.test.js]: it does not need done. it's an async not promise, that why its they is an await in the front @albert85

File: src/modules/guestHouse/__tests__/updateGuestHouse.test.js (hunk 4 above)
Comment (Spectrumsun) [reply src/modules/guestHouse/__tests__/updateGuestHouse.test.js]: it does not need done. it's an async not promise, that why they is an await in the front @albert85

File: src/modules/centers/__tests__/changeCenter.test.js [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+    fullName: 'white windows ',
+    email: 'white.windows@andela.com'
+  }
+};
+
+const token = Utils.generateTestToken(payload);
+const token2 = Utils.generateTestToken(payload2);
+
+describe('Update center test', () => {
+  beforeAll(async () => {
```
[the remark this answers is not in the document]
Comment (Spectrumsun) [reply src/modules/centers/__tests__/changeCenter.test.js]: it does not need done. it's an async not promise, that why they is an await in the front @albert85

**Comment by Spectrumsun** [comment]
Done @daddychukz

**Comment by Spectrumsun** [comment]
@daddychukz have the removed it

**Comment by daddychukz** [comment]
All's good @samuelkubai

**Pull request merged**